app/models/room.rb

```diff
@@ -1,3 +1,13 @@
 class Room < ApplicationRecord
    belongs_to :hotel
+   has_many :guest_rooms
+   has_many :guests, through: :guest_rooms
+
+   def guest_count
+      self.guests.count
+   end
+
+   def guest_names
+      self.guests.pluck(:name)
+   end
 end
```

Comment:
`Then I see a unique list of all guests that have stayed at this hotel`

Just taking a look at that line in the user story, it tells me that we want a list of unique guests for any given hotel. So instead of trying to add a method in the `Room` model, I would be adding it to the `Hotel` model. 

If we take a look at the `Hotel` model, we'll see that we have one association between a `Hotel` and a `Room`. We also know that if we take a look at the `Room` model, a `Room` has an association with `Guest`s through `GuestRoom`.

There's a neat trick in Rails that you can use those predefined associations to simplify our joins.

```ruby
class Hotel < ApplicationRecord
  has_many :rooms
  has_many :guests, through: :rooms # We are gaining access to guests of this hotel through the rooms of this hotel

  def all_guests
    guests.distinct
  end
end
```

Rails magic FTW. Play around with this and see what SQL gets printed in the terminal. A multi-join!